src/main/java/ikura/Main.java

```diff
@@ -0,0 +1,28 @@
+// Main.java
+// Copyright (c) 2020, zhiayang, Apache License 2.0.
+
+package ikura;
+
+import java.util.Scanner;
+import java.io.IOException;
+
+import ikura.Bot;
+import ikura.Frontend;
+import ikura.Database;
+import ikura.TaskList;
+
+public class Main {
+
+    private static final String BOT_NAME  = "ikurabowl";
+    private static final String DB_PATH   = "data/tasks.txt";
+
+    public static void main(String[] args) {
+        var tasks = new TaskList(new Database(DB_PATH));
+        var ui = new Frontend(BOT_NAME);
+        var bot = new Bot(ui, tasks);
+
+        ui.greet();
+        while (ui.readLine().map(bot::processCommand).orElse(false))
+            ;
```

Comment:
Perhaps the ; could be in line 25?

Reply:
tbh i think the style guide would say that i just make an empty block with `{ }`.

anyway (style guide aside) my motivation for putting the semicolon on a separate line for empty loop bodies is that it makes it explicit that the body is empty, because it can be easy to miss the semicolon at the end of the line and think that the proceeding statement is part of the loop.